oscore-cf/src/main/java/org/eclipse/californium/oscore/OptionJuggle.java

```diff
@@ -0,0 +1,328 @@
+		allEOptions.add(OptionNumberRegistry.SIZE1);
+		return allEOptions;
+	}
+
+	/**
+	 * Prepare a set or original CoAP options for unprotected use with OSCore.
+	 * 
+	 * @param options the original options
+	 * @param osOpt the OSCore option
+	 * 
+	 * @return the OSCore-U option set
+	 */
+	public static OptionSet prepareUoptions(OptionSet options) {
+		boolean hasProxyUri = options.hasProxyUri();
+		boolean hasUriHost = options.hasUriHost();
+		boolean hasUriPort = options.hasUriPort();
+		boolean hasProxyScheme = options.hasProxyScheme();
+		boolean hasMaxAge = options.hasMaxAge();
+		boolean hasObserve = options.hasObserve();
+
+		OptionSet ret = new OptionSet();
+
+		if (hasUriHost) {
+			ret.setUriHost(options.getUriHost());
+		}
+
+		if (hasUriPort) {
+			ret.setUriPort(options.getUriPort());
+		}
+
+		if (hasMaxAge) {
+			ret.setMaxAge(options.getMaxAge());
+		}
+
+		if (hasProxyScheme) {
+			ret.setProxyScheme(options.getProxyScheme());
+		}
+
+		if (hasObserve) {
+			ret.setObserve(options.getObserve());
+		}
+
+		if (hasProxyUri) {
+			String proxyUri = options.getProxyUri();
+			proxyUri = proxyUri.replace("coap://", "");
+			proxyUri = proxyUri.replace("coaps://", "");
+			int i = proxyUri.indexOf('/');
+			if (i >= 0) {
+				proxyUri = proxyUri.substring(0, i);
+			}
+			proxyUri = "coap://" + proxyUri;
+			ret.setProxyUri(proxyUri);
+		}
+
+		byte[] oscore = options.getOscore();
+		if (oscore != null) {
+			ret.setOscore(oscore);
+		}
+
+		return ret;
+	}
+
+	/**
+	 * Prepare a set or original CoAP options for encryption with OSCore.
+	 * 
+	 * @param options the original CoAP options
+	 * 
+	 * @return the option to be encrypted
+	 */
+	public static OptionSet prepareEoptions(OptionSet options) {
+		OptionSet ret = new OptionSet();
+
+		for (Option o : options.asSortedList()) {
+			switch (o.getNumber()) {
+
+			// We do not support observe as of now and so the message is
+			// processed as a normal message
+			case OptionNumberRegistry.OBSERVE:
+				options.removeObserve();
+				break;
+			case OptionNumberRegistry.URI_HOST:
+			case OptionNumberRegistry.URI_PORT:
+			case OptionNumberRegistry.PROXY_SCHEME:
+			case OptionNumberRegistry.OSCORE:
+				// do not encrypt
+				break;
+			case OptionNumberRegistry.PROXY_URI:
+				// create Uri-Path and Uri-Query
+				String proxyUri = o.getStringValue();
+				proxyUri = proxyUri.replace("coap://", "");
+				proxyUri = proxyUri.replace("coaps://", "");
+				int i = proxyUri.indexOf('/');
+				if (i >= 0) {
+					proxyUri = proxyUri.substring(i + 1, proxyUri.length());
+				} else {// No Uri-Path and Uri-Query
+					break;
+				}
+				i = proxyUri.indexOf("?");
+				String uriPath = proxyUri;
+				String uriQuery = null;
+				if (i >= 0) {
+					uriPath = proxyUri.substring(0, i);
+					uriQuery = proxyUri.substring(i + 1, proxyUri.length());
+				}
+
+				if (uriPath != null) {
+					String[] uriPaths = uriPath.split("/");
+					for (int idx = 0; idx < uriPaths.length; idx++) {
```

Comment:
`OptionSet.setUriPath` split and adds the single options for each path element from the provided path. It obsoletes the split and for loop here. 
So just replace line 169 to 172 with the
`ret.setUriPath(uriPath);`

Reply:
Sorry, my mistake.